src/main/webapp/resources/js/pages/sequencing-runs/components/SequencingRunDetailsPage.jsx

```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import { useParams } from "react-router-dom";
+import {
+  useGetSequencingRunDetailsQuery,
+  useGetSequencingRunFilesQuery
+} from "../../../apis/sequencing-runs/sequencing-runs";
+import { Button, Col, Row, Table, Typography } from "antd";
+import { formatDate } from "../../../utilities/date-utilities";
+import { setBaseUrl } from "../../../utilities/url-utilities";
+import { IconDownloadFile } from "../../../components/icons/Icons";
+import { LinkButton } from "../../../components/Buttons/LinkButton";
+import { BasicList } from "../../../components/lists";
+import { NarrowPageWrapper } from "../../../components/page/NarrowPageWrapper";
+import { SequencingRunStatusBadge } from "./SequencingRunStatusBadge";
+
+/**
+ * React component to display the sequencing run details page.
+ * @returns {*}
+ * @constructor
+ */
+export default function SequencingRunDetailsPage() {
+  const ADMIN_SEQUENCE_RUNS_URL = "admin/sequencing-runs";
+  const isAdmin = window.TL._USER.systemRole === "ROLE_ADMIN";
+  const showBack = isAdmin && document.referrer.includes(ADMIN_SEQUENCE_RUNS_URL);
+  const {runId} = useParams();
+  const goToAdminSequenceRunListPage = () =>
+    (window.location.href = setBaseUrl(ADMIN_SEQUENCE_RUNS_URL));
+  const {
+    data: run = {},
+    isLoading: isRunLoading
+  } = useGetSequencingRunDetailsQuery(runId);
+  const {
+    data: files = [],
+    isLoading: isFilesLoading
+  } = useGetSequencingRunFilesQuery(runId);
+  const detailsList = isRunLoading
+    ? []
+    : [
+      {
+        title: i18n("SequencingRunDetailsPage.list.sequencerType"),
+        desc: run.sequencerType
+      },
+      {
+        title: i18n("SequencingRunDetailsPage.list.uploadStatus"),
+        desc: <SequencingRunStatusBadge status={run.uploadStatus}/>
+      },
+      {
+        title: i18n("SequencingRunDetailsPage.list.uploadUser"),
+        desc: isAdmin ? (<LinkButton
+            text={run.userName}
+            href={setBaseUrl(`/users/${run.userId}`)}/>
+        ) : run.userName
+      },
+      {
+        title: i18n("SequencingRunDetailsPage.list.createdDate"),
+        desc: formatDate({date: run.createdDate})
+      },
+      {
+        title: i18n("SequencingRunDetailsPage.list.description"),
+        desc: run.description
+      },
+    ];
+  const optionalPropertiesList = [];
```

Comment:
This would probably be better written:

```javascript
const optionalPropertiesList = run.optionalProperties ? 
  Object.entries(run.optionalProperties)
        .map(([key, value]) => ( optionalPropertiesList.push({
            title: key, 
          desc: value}))) : 
  [];
```

Reply:
This fails for me with the following error, "Uncaught ReferenceError: Cannot access 'optionalPropertiesList' before initialization".